From 92c55802144a024fb48ca5babeb99254209d2c71 Mon Sep 17 00:00:00 2001
From: Nils Philippsen <nils@redhat.com>
Date: Thu, 24 Oct 2013 14:29:25 +0200
Subject: [PATCH] patch: hwdb

Squashed commit of the following:

commit 6686fded6523a04b26d02e7bbeb906a579c9ef5f
Author: Nils Philippsen <nils@redhat.com>
Date:   Thu Oct 24 14:22:17 2013 +0200

    sane-desc: fix udev hwdb generation

    Generated hwdb files listed the vendor ID instead of the product ID
    which made udevd not recognizing scanner devices as such. Thanks to
    Fabrice Bellet who spotted the problem.

    (cherry picked from commit 3b96baef65ea6c315937f5cd2253c6b6c62636d8)
---
 tools/sane-desc.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/tools/sane-desc.c b/tools/sane-desc.c
index df6d348..ae218c3 100644
--- a/tools/sane-desc.c
+++ b/tools/sane-desc.c
@@ -3827,7 +3827,7 @@ print_hwdb (void)
 
       for(j = 0; j < 4; j++) {
         vendor_id[j] = toupper(vendor_id[j]);
-        product_id[j] = toupper(vendor_id[j]);
+        product_id[j] = toupper(product_id[j]);
       }
 
       printf ("usb:v%sp%s*\n libsane_matched=yes\n\n",
-- 
1.8.3.1

